indico/htdocs/js/indico/modules/timetable/timetable/Management.js

```diff
@@ -1068,3 +1068,84 @@ type("FitInnerTimetableDialog", ["ConfirmPopup"], {
         this.ConfirmPopup($T("Fit timetable to content"), this.__getContent(), this.__handler);
     }
 );
+
+
+type("SessionSectionPopupMenu", ["SectionPopupMenu"],
+    {
+        _processItem: function(pair) {
+            var self = this;
+            var value = pair.get();
+            var color = null;
+            var title = null;
+
+            if(exists(value.title)){
+                title = value.title;
+            }else {
+                title = pair.key;
+            }
+
+            if(exists(value.color)){
+                color = value.color;
+                value = value.func;
+            }
+
+            var colorSquare = null;
+            if(color !== null){
+                colorSquare = Html.div({style:{backgroundColor: color, color: color, cssFloat: 'right', width: '15px', height:'15px'}});
+            }
+
+            var link = Html.a({className:'fakeLink', style:{display: 'inline', padding: 0, paddingLeft: '4px', paddingRight: '4px'}}, Util.truncate(title));
+            var divInput = Html.div({style:{height:'20px', overflow:'auto'}}, colorSquare, link);
+
+            if(typeof value == "string" ) {
+                link.setAttribute('href', value);
+                if (self.closeOnClick) {
+                    link.observeClick(function() {
+                        self.close();
+                    });
+                }
+            }
+            else {
+                link.observeClick(value.PopupWidget?
+                                  function(e) {
```

Comment:
you moved it around so the diff is already big --> can you fix the awful indentation?